### DISCUSSION TITLE
When I switch from the default user to root, vscode cannot be started

### DISCUSSION THREAD

**Discussion by Zcaic** [microsoft/WSL#9873] [General]
### Windows Version

Microsoft Windows [Version 10.0.19045.2728]

### WSL Version

1.1.3.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1

### Distro Version

Ubuntu22.04

### Repro Steps

1. When I start wsl as the default user, wsl has a Windows environment variable and can start vscode.
2. However, when I switch to root, the Windows environment variables are missing, so vscode cannot be started. 
3. When I switch back to the user directory, the Windows environment variables are still missing. So that I could not start vscode either.

### Expected Behavior

I want to know how to preserve or load Windows environment variables after I switch users.

### Actual Behavior

Windows environment variables have disappeared, so that vscode cannot be started.

**Comment by ghost** [comment]
@Zcaic, can you be a little more specific with your repro instructions? Are you launching via `wsl -u root` ? Thanks!

**Comment by Zcaic** [comment]
@pmartincic 
No，I set the default user， When I start wsl, I am the default user， vscode can be started through the "code ." command. But after I use "sudo su" to switch to root， Windows environment variables are missing, so I cannot start vscode from root. When I switch back to the user, vscode still cannot be started through " code . "

**Comment by elsaco** [comment]
@Zcaic try `sudo -E su` to pass your environment to root. From the manual:

```
     -E, --preserve-env
             Indicates to the security policy that the user wishes to preserve their existing environment variables.
             The security policy may return an error if the user does not have permission to preserve the environ-
             ment.

```

**Comment by Zcaic** [comment]
@elsaco It's useful for me, thanks a lot !
